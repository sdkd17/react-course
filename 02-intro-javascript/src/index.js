

const apiKey = 'nJgFoWKmfae1ZgLV4aOMdPlNp4VsVJVS';
const tag = 'NBA';
const rating = 'g';

const peticion = fetch(`https://api.giphy.com/v1/gifs/random?api_key=${apiKey}&tag=${tag}&rating=${rating}`);

// peticion.then ( (resp) => {
//   resp.json().then( ( data ) => {
//     console.log(data);
//   })
// })
// .catch( console.wanr );

peticion
  .then ( resp => resp.json())
  .then ( ({ data } ) => {
    console.log(data);
    const { url } = data.images.original;
    console.log(url)

    const img = document.createElement('img');
    img.src = url;


    document.body.append(img);
  })
.catch( console.wanr );

