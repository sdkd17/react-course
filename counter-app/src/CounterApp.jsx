import PropType from 'prop-types';
import { useState } from 'react';



export const CounterApp = ( {value} ) => {

  const [counter, setCounter] = useState( value );

  const handleAdd = () => {
    setCounter((c) => c+1 );
  }
  
  const handleSub = () => {
    setCounter((c) => c > 0 ? c = c-1 : c );
  }
  
  const handleReset = () => {
    setCounter((c) => value );
  }
 

  return (
    <>
      <h1>CounterApp</h1>
      <h2> { counter } </h2>

      <button onClick={ handleAdd }>+1</button>
      <button onClick={ handleSub }>-1</button>
      <button aria-label='btn-reset' onClick={ handleReset }>reset</button>
      
    </>
  )
}

CounterApp.propTypes = {
  value: PropType.number.isRequired,
}


