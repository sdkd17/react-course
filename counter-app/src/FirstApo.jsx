import PropTypes from 'prop-types';

export const FirstApo = ({title, subTitle}) => {
  return (
    <>
      <h1>{ title }</h1>

      <h5> { subTitle } </h5>
    </>
  )
}

FirstApo.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
}
