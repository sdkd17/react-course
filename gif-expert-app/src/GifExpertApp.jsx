import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

function camelize(str) {
  return str.replace(/(?:^\w|[A-Z]|\b\w|\s+)/g, function(match, index) {
    if (+match === 0) return ""; // or if (/\s+/.test(match)) for white spaces
    return index === 0 ? match.toLowerCase() : match.toUpperCase();
  });
}

export const GifExpertApp = () => {

	const [categories, setCategories] = useState( [] );

	const onAddCategory = (newCategory) => {
		if( categories.includes(newCategory) ) return;
		
		setCategories( [...categories, newCategory] );
		// setCategories( cat => [...cat, 'newCat'] );
	}

	return (
		<>
			<h1>GifExpertApp</h1>

			<AddCategory 
				// setCategories={ setCategories }
				onNewCategory={ onAddCategory }
			/>
			{/* set categories manda categories como parametro implicito */}
			
			<ol>
				{ 
					categories.map( category => {
						return <li key={category}>{ camelize(category) }</li>
					}		
				)}
			</ol>
		</>
	)
}
