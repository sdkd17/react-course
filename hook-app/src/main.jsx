import React from 'react';
import ReactDOM from 'react-dom/client';
import { Layout } from './05-useLayoutEffect/layout';
// import { FocusScreen } from './04-useRef/FocusScreen';
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks';
// import { CounterApp } from './01-useState/CounterApp';
// import { HooksApp } from './HooksApp';
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook';
// import { SimpleForm } from './02-useEffect/SimpleForm';
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook';

import './index.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <Layout />
    {/* <FocusScreen /> */}
    {/* <MultipleCustomHooks /> */}
    {/* <FormWithCustomHook /> */}
    {/*<SimpleForm />*/}
    {/* <CounterWithCustomHook /> */}
    {/* <CounterApp /> */}
    {/* <HooksApp /> */}
  </React.StrictMode>
)
