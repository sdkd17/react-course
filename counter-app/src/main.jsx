import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApo } from './FirstApo';

import './styles.css'


ReactDOM.createRoot( document.getElementById( 'root' ) ). render (
  <React.StrictMode>
    {/* <FirstApo title={"Titulo"} subTitle={"subtitulo"}/> */}
    <CounterApp value={100}/>
  </React.StrictMode>
)