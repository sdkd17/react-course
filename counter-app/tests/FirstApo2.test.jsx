import { render, screen } from '@testing-library/react';
import { FirstApo } from '../src/FirstApo';

describe('Pruebas en FirtApo 2', () => { 

  const title = 'Hola, soy Sergio';
  const subTitle = 'subtitle';

  test('debe de hacer match con el sanpshot', () => { 
    
    const { container } = render( <FirstApo title={title}/> );

    expect( container ).toMatchSnapshot();

  });

  test('debe mostrar el mensaje "Hola, soy Sergio"', () => { 
    render( <FirstApo title={title}/> );
    expect( screen.getByText(title) ).toBeTruthy();
  });

  test('should mostrar titulo en un h1', () => { 
    render( <FirstApo title={title}/> );
    expect( screen.getByRole('heading', {level: 1}).innerHTML ).toContain(title);
  });

  test('should mostar subtitulo enviado por props', () => { 
    render( <FirstApo title={title} subTitle={subTitle}/> );
    expect( screen.getByText(subTitle) ).toBeTruthy();
  })
})