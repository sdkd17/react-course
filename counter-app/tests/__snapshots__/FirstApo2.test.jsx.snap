// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirtApo 2 debe de hacer match con el sanpshot 1`] = `
<div>
  <h1>
    Hola, soy Sergio
  </h1>
  <h5>
     
     
  </h5>
</div>
`;
